nats/src/main.rs

```diff
@@ -168,21 +258,46 @@ impl NatsMessagingProvider {
             // up with 20 subscriptions all getting slammed with up to 75 tasks, but we should wait
             // to do anything until we see what happens with real world usage and benchmarking
             let semaphore = Arc::new(Semaphore::new(75));
-            while let Some(msg) = subscription.next().await {
-                let span = tracing::debug_span!("handle_message", actor_id = %link_def.actor_id);
-                span.in_scope(|| {
-                    wasmbus_rpc::otel::attach_span_context(&msg);
-                });
-                let permit = match semaphore.clone().acquire_owned().await {
-                    Ok(p) => p,
-                    Err(_) => {
-                        warn!("Work pool has been closed, exiting queue subscribe");
-                        break;
+
+            // Listen for another NATS message to come in
+            loop {
```

Comment:
So I actually think you might be able to get away with storing the `join_handle` for this spawned task and aborting all of them when the link is deleted. I _think_ it cleans up the underlying future and `Drop`s everything. I did double check and the `Subscription` struct does clean itself up on drop. What I think was causing the bug is that we never stopped this thread. If for some reason it doesn't clean up a single broadcast sender and a cloned receiver is all you'll need (as mentioned in the previous comment). 

Either way, the `actors` data member can be something like `actors: Arc<RwLock<HashMap<String,(Client, Vec<JoinHandle>)>>>,` (or putting the client and list of `JoinHandle`s in a struct). Then you can push a join handle on to that vec.

Reply:
I'm not sure what happens when multiple links listen on the same channel, but it looks like we'd get two separate loops on the same channel. I wouldn't want deleting one link to stop all of those right?

Didn't thin of changing `actors` though -- will consider that

Reply:
refactored in favor of holding `JoinHandle`s